common/tempstorage/memstore/storage.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 FoxyUtils ehf. All rights reserved.
+//
+// Use of this source code is governed by the terms of the Affero GNU General
+// Public License version 3.0 as published by the Free Software Foundation and
+// appearing in the file LICENSE included in the packaging of this file. A
+// commercial license can be purchased on https://unidoc.io.
+
+// Package memstore implements tempStorage interface
+// by using memory as a storage
+package memstore
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"sync"
+
+	"github.com/unidoc/unioffice/common/tempstorage"
+)
+
+// memStorage contains and manages memdataCell items as operating system manages files
+type memStorage struct {
+	m    map[string]*memDataCell
```

Comment:
a sync.Map might also work?

Reply:
looks like it is more suitable as from documentation:

`The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.`

The case 1 looks like ours, so I changed to sync.Map.